src/services/identity/SitesService.ts

```diff
@@ -78,6 +84,107 @@ class SitesService {
   }
 
+  async getCommitAuthorEmail(commit: GitHubCommitData) {
+    const { message } = commit
+
+    // Commit message created as part of phase 2 identity
+    if (message.startsWith("{") && message.endsWith("}")) {
+      const { userId }: { userId: string } = JSON.parse(message)
+      const user = await this.usersService.findById(userId)
+
+      if (user) {
+        return user.email
+      }
+    }
+
+    // Legacy style of commits, or if the user is not found
+    const {
+      author: { email: authorEmail },
+    } = commit
+    return authorEmail
+  }
+
+  async getUrlsOfSite(
+    sessionData: UserWithSiteSessionData,
+    urlTypes: SiteUrlTypes[]
+  ): Promise<SiteUrls> {
+    // Tries to get the site urls in the following order:
+    // 1. From the deployments database table
+    // 2. From the config.yml file
+    // 3. From the GitHub repository description
+    // Otherwise, returns an empty string for the url
+    const { siteName } = sessionData
+    let configYmlData: ConfigYmlData = {}
+    let repoDescTokens: string[] = []
+
+    const site = await this.siteRepository.findOne({
+      where: { name: siteName },
+      include: {
+        model: Deployment,
+        as: "deployment",
+      },
+    })
+
+    if (!site) {
+      return {
+        staging: "",
+        prod: "",
+      }
+    }
+
+    const deployment = site?.deployment
+    const output: SiteUrls = {
+      staging: deployment?.stagingUrl ? deployment.stagingUrl : "",
+      prod: deployment?.productionUrl ? deployment.productionUrl : "",
+    }
+
+    for (const urlType of urlTypes) {
+      // We only call ConfigYmlService only when necessary
+      if (!output[urlType] && Object.keys(configYmlData).length === 0) {
+        const {
+          content: configData,
+        }: { content: ConfigYmlData } = await this.configYmlService.read(
+          sessionData
+        )
+
+        configYmlData = configData
+      }
+
+      if (!output[urlType] && urlType in configYmlData) {
+        const configSiteUrl = configYmlData[urlType]
+
+        if (configSiteUrl) {
+          output[urlType] = configSiteUrl
+        }
+      }
+
+      // We only call GitHubService only when necessary
+      if (!output[urlType] && repoDescTokens.length == 0) {
+        const {
+          description,
+        }: { description: string } = await this.gitHubService.getRepoInfo(
+          sessionData
+        )
+
+        // Retrieve the url from the description
+        // repo descriptions have varying formats, so we look for the first link
+        repoDescTokens = description.replace("/;/g", " ").split(" ")
+      }
+
+      if (!output[urlType]) {
+        // Site urls also contain the site url types in their url
+        const siteUrl = repoDescTokens.find(
+          (token) => token.includes("http") && token.includes(urlType)
+        )
+        if (siteUrl) {
+          output[urlType] = siteUrl
+        }
+      }
+    }
+
+    return output
```

Comment:
note that `output` here is not guaranteed to be well-formed. legacy sites could potentially have misbehaving behaviour and hence, still have undefined `staging | prod`. 

at the end, we could validate properties and return either 1. well formed output, 2. a single output (either `staging | prod` only or 3. error

double check wrt github repo desc @alexanderleegs

Reply:
Adjusted in 4432732. If all 3 sources of data does not contain the info we need, then we will return an UnprocessableError.